core/node/node_framework/src/implementations/layers/state_keeper/main_batch_executor.rs

```diff
@@ -32,17 +35,35 @@ impl WiringLayer for MainBatchExecutorLayer {
     async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
         let master_pool = context.get_resource::<MasterPoolResource>().await?;
 
-        let builder = MainBatchExecutor::new(
-            self.db_config.state_keeper_db_path,
+        let (storage_factory, task) = AsyncRocksdbCache::new(
             master_pool.get_singleton().await?,
+            self.db_config.state_keeper_db_path,
+            self.state_keeper_config.enum_index_migration_chunk_size(),
+        );
+        let builder = MainBatchExecutor::new(
+            Arc::new(storage_factory),
             self.state_keeper_config.max_allowed_l2_tx_gas_limit.into(),
             self.state_keeper_config.save_call_traces,
             self.state_keeper_config.upload_witness_inputs_to_gcs,
-            self.state_keeper_config.enum_index_migration_chunk_size(),
             false,
         );
 
         context.insert_resource(BatchExecutorResource(Unique::new(Box::new(builder))))?;
+        context.add_task(Box::new(RocksdbCatchupTask(task)));
+        Ok(())
+    }
+}
+struct RocksdbCatchupTask(AsyncCatchupTask);
```

Comment:
Nit: Missing `Debug` implementation and an blank line before the type declaration.